### PR TITLE
tchannel/inbound: tests

### PR DESCRIPTION
**Pull request by abhinav**
Moved from #65.

[Review on Reviewable](https://reviewable.io/reviews/yarpc/yarpc-go/70)

### CODE REVIEW

**Comment by abhinav** [comment]
This is ready for eyes

**Comment by abhinav** [comment]
This is ready for eyes. CC @yarpc/golang

File: transport/tchannel/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	"golang.org/x/net/context"
+)
+
+func TestHandlerFailures(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
```
Comment (breerly) [inline transport/tchannel/handler_test.go]: damn son, this is a serious table test :+1:

**Comment by breerly** [comment]
:shipit:

**Comment by prashantv** [comment]
Review status: 0 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[transport/tchannel/handler_test.go, line 41 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCNz22PqR2R96lXCwls:-KCNz22PqR2R96lXCwlt:2030583985) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/handler_test.go#L41)):*
optional: should we initialize the mock controller for each case, to make sure the handler was only called when you actually expected it vs being called from any of the cases

---

*[transport/tchannel/handler_test.go, line 63 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO0voXlNv53ULEsZpD:-KCO0voYeWp5-wS4MdUY:1407304068) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/handler_test.go#L63)):*
are things like empty service, empty caller, or empty method considered errors?

---

*[transport/tchannel/inbound_test.go, line 59 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO1Q3VcqD5CPtGt-48:-KCO1Q3VcqD5CPtGt-49:666335207) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/inbound_test.go#L59)):*
can you test the Addr after it starts listening looks like what you'd expect

---

*[transport/tchannel/inbound_test.go, line 80 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO1rF1bEKjAeODs51o:-KCO1rF2rmjyRBXWOjKl:1584916) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/inbound_test.go#L80)):*
make sure addr is set correctly

---

*[transport/tchannel/outbound_test.go, line 77 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO2k4_hUz1YQeG-xxL:-KCO2k4_hUz1YQeG-xxM:-104284449) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/outbound_test.go#L77)):*
Why do you need leeway? The incoming request TTL will always be <= 200ms, and then the context is created with a deadline at some earlier point, so it should definitely be <- 200ms by now right?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/yarpc-go/70)*

**Comment by abhinav** [comment]
Review status: 0 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[transport/tchannel/handler_test.go, line 41 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCNz22PqR2R96lXCwls:-KCRg2PAAG9RK5XIIQ0s:1074753703) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/handler_test.go#L41)):*
Makes sense. Done. `Finish()` won't be called on the mock now unless the test succeeds, which is okay because unexpected calls fail anyway.

---

*[transport/tchannel/handler_test.go, line 63 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO0voXlNv53ULEsZpD:-KCRgLmqbpSDhg7C15Xq:-1666932296) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/handler_test.go#L63)):*
Yes, but aren't those required by the TChannel protocol? The handler doesn't even validate their presence right now.

---

*[transport/tchannel/inbound_test.go, line 59 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO1Q3VcqD5CPtGt-48:-KCRhQhHbGtZ_CLoMG9f:-1661074918) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/inbound_test.go#L59)):*
Discussed offline. Addr not exposed anymore.

---

*[transport/tchannel/inbound_test.go, line 80 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO1rF1bEKjAeODs51o:-KCRhSNyGQ5AJe7moNAP:-1661074918) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/inbound_test.go#L80)):*
Discussed offline. Addr not exposed anymore.

---

*[transport/tchannel/outbound_test.go, line 77 \[r1\]](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-KCO2k4_hUz1YQeG-xxL:-KCRhxbN2Dww66jnfmTk:-1813946197) ([raw file](https://github.com/yarpc/yarpc-go/blob/8211ee2cf6cdfbc6dc9cc870f3a331fd3c15b9c2/transport/tchannel/outbound_test.go#L77)):*
Oh right. I forgot how to math.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/yarpc-go/70)*

**Comment by prashantv** [comment]
:lgtm: 

---

Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/yarpc-go/70#-:-KCRjmkSuLSW1kOkMyFx:980077423)*

**Pull request merged**